portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +36,57 @@
 public class DataServlet extends HttpServlet {
 
   private List<String> messages;
-  
-  @Override 
-  public void init() {
-    messages = new ArrayList<String>(); 
-    messages.add("Hello, this is my first comment");
-    messages.add("Hello, this is my second comment");
-    messages.add("Hi, this is my third comment");
-  }
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String jsonResponse = convertToJson(messages);
+    Query query = new Query("Recommendation").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    //Number limit for entities loaded
+    int maxNumber = results.countEntities();
+    try {
+      maxNumber = Integer.parseInt(request.getParameter("max"));
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + request.getParameter("max")); 
+    }
+    
+    List<Entity> resultsList = results.asList(FetchOptions.Builder.withLimit(100));
+    List<String> myToDos = new ArrayList<>(); 
+
+    for (int i = 0; i < maxNumber; i++) {
+      Entity entity = resultsList.get(i);
+      String name = (String) entity.getProperty("name");
+      String category = (String) entity.getProperty("category");
+      String content = (String) entity.getProperty("content");
+      String comment = (String) entity.getProperty("comment");
```

Comment:
Datastore field names appear twice in this file. It's recommended to add a class level constant variable to ensure consistency between get and post.

Reply:
okay, fixed!